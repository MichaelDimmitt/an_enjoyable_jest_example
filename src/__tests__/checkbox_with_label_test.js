// Copyright 2004-present Facebook. All Rights Reserved.

/* eslint-disable no-unused-vars */

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import * as TestUtils from 'react-dom/test-utils';
import CheckboxWithLabel from '../CheckboxWithLabel';
import context from 'jest-plugin-context';

it('renders correctly', () => {
    const tree = renderer
      .create(<CheckboxWithLabel labelOn="On" labelOff="Off"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});
describe('a sane universe', () => {
  context('planet earth', () => {

    it('CheckboxWithLabel changes the text after click', () => {
      // Render a checkbox with label in the document
      const checkbox = TestUtils.renderIntoDocument(
        <CheckboxWithLabel labelOn="On" labelOff="Off" />
      );

      const checkboxNode = ReactDOM.findDOMNode(checkbox);

      // Verify that it's Off by default
      expect(checkboxNode.textContent).toEqual('Off');

      // Simulate a click and verify that it is now On
      TestUtils.Simulate.change(
        TestUtils.findRenderedDOMComponentWithTag(checkbox, 'input')
      );
      expect(checkboxNode.textContent).toEqual('On');
    });
  })
})
