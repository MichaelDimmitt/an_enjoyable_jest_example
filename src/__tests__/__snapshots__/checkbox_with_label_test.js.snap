// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    Off
  </label>
</div>
`;
